src/Components/WebAssembly/WebAssembly/src/Prerendering/WebAssemblyComponentParameterDeserializer.cs

```diff
@@ -3,12 +3,18 @@
 
 using System.Diagnostics.CodeAnalysis;
 using System.Text.Json;
+using System.Text.Json.Serialization;
 using static Microsoft.AspNetCore.Internal.LinkerFlags;
 
 namespace Microsoft.AspNetCore.Components;
 
 internal sealed class WebAssemblyComponentParameterDeserializer
 {
+    private static readonly JsonSerializerOptions _jsonSerializerOptions = new(WebAssemblyComponentSerializationSettings.JsonSerializationOptions)
```

Comment:
Depending on how `WebAssemblyComponentSerializationSettings.JsonSerializationOption` is defined, you might be able to avoid creating this intermediate options instance by configuring your `JsonSerializerContext` via the `JsonSourceGenerationsOptionsAttribute`.